### ISSUE TITLE
problem to compile micmac in ubuntu 17.10

### ISSUE THREAD

**Issue by OpenHDZ** [micmacIGN/micmac#61]
I tray to compile micmac and i fallow all the installation instruction but i have this response when I use cmake

-- The C compiler identification is GNU 7.2.0
-- The CXX compiler identification is GNU 7.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenGL: /usr/lib/i386-linux-gnu/libGL.so  
-- -----> OpenGL found
-- using Qt 5.9.1
-- QT tools enabled (Qt5)
c++11
 -std=c++11
-- Found Git: /usr/bin/git (found version "2.14.1") 
-- -- Git executable found
-- Derniere revision git de MicMac : 
-- Looking for XOpenDisplay in /usr/lib/i386-linux-gnu/libX11.so;/usr/lib/i386-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/i386-linux-gnu/libX11.so;/usr/lib/i386-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib/i386-linux-gnu/libX11.so
CMake Warning (dev) at /usr/share/cmake-3.9/Modules/FindOpenMP.cmake:200 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "c" will no longer be dereferenced when the policy is
  set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /usr/share/cmake-3.9/Modules/FindOpenMP.cmake:324 (_OPENMP_GET_FLAGS)
  CodeExterne/Poisson/CMakeLists.txt:6 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- PoissonRecon/SurfaceTrimmer: OpenMP enabled
-- Configuring done
CMake Error at src/CMakeLists.txt:174 (add_library):
  Cannot find source file:

    uti_phgrm/NewOri/GpsLoc.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx

CMake Error: Cannot determine link language for target "elise".
CMake Error: CMake can not determine linker language for target: elise
-- Generating done
-- Build files have been written to: /home/wroot/projet_scan/micmac-master/build
wroot@wroot-HP-ProBook-4540s:~/projet_scan/micmac-master/build$ cmake ../
-- -----> OpenGL found
-- using Qt 5.9.1
-- QT tools enabled (Qt5)
c++11
 -std=c++11
-- -- Git executable found
-- Derniere revision git de MicMac : 
-- PoissonRecon/SurfaceTrimmer: OpenMP enabled
-- Configuring done
CMake Error at src/CMakeLists.txt:174 (add_library):
  Cannot find source file:

    uti_phgrm/NewOri/GpsLoc.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx

CMake Error: Cannot determine link language for target "elise".
CMake Error: CMake can not determine linker language for target: elise
-- Generating done
-- Build files have been written to: /home/wroot/projet_scan/micmac-master/build

**Comment by erupnik** [comment]
Hi,
it should be working now. Git pull to update the code and re-compile;
e

**Comment by alexmyczko** [comment]
Do you have plans to create an official ubuntu package?

**Comment by rjanvier** [comment]
Hello again Alex,
I worked on that and it worked at some point. In some aspects, It's "easy" since there is not so much dependencies in MicMac (buildeps : QT, CUDA, openCL ar optionals, install deps: make, imagemagick, exiv2, exiftool). But, there are some issues that needs to be tackled.
- FHS: MicMac uses some non standard path w.r.t FHS policy. It could be tackled by quilt patches without being invasive (like I did)
- Trim out Shewchuk's triangle out of the source code. It's licensed under a "research only" license that is not allowed in Debian repos (it's the cause why my CloudCompare package you now maintain was rejected few years ago and now we use cgal)
- Trim out SIFT binaries and sift source code out of the archive too, because it's patented a replace it by something more DFSG friendly (Akaze from opencv for example) @luca-penasa worked on some integration with opencv  some times ago https://github.com/luca-penasa/tarapio and I know that MPD (@deseilligny) is working right now on a new detector/escriptor implementation, I don't know how far it's related to patented ones (SIFT, SURF).
